core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -45,13 +47,20 @@
 
     private final String fuzziness;
 
+    /** the prototype constant is intended for deserialization when used with
+     * {@link org.elasticsearch.common.io.stream.StreamableReader#readFrom(StreamInput)} */
+    static final Fuzziness PROTOTYPE = AUTO;
+    
     private Fuzziness(int fuzziness) {
         Preconditions.checkArgument(fuzziness >= 0 && fuzziness <= 2, "Valid edit distances are [0, 1, 2] but was [" + fuzziness + "]");
         this.fuzziness = Integer.toString(fuzziness);
     }
 
     private Fuzziness(String fuzziness) {
-        this.fuzziness = fuzziness;
+        if (fuzziness == null) {
```

Comment:
i see it throws an NPE exception, not as neat as the error we return to the user, unless you want to wrap the call in a try catch and still return the nicer error to the user.